### PR TITLE
Source MySQL\MsSql\Postgres: added RDS base performance tests

### PR DESCRIPTION
**Pull request by etsybaev**
## What
Need to test benchmarks

## How
Added tests for MySql, MsSql and Postgres RDS DBs.

## 🚨 User Impact 🚨
No user impacts, tests only

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

**Review by tuliren — COMMENTED** [review]
This is great!

Currently in `AbstractSourcePerformanceTest`, the parameters to create the db are instance level variables. Each test method and the db creation method use those parameters to communicate indirectly about what db to create. This seems a strange pattern. What about removing those instance level variables, and pass them in directly in the test methods?

**Comment by etsybaev** [comment]
> This is great!
> 
> Currently in `AbstractSourcePerformanceTest`, the parameters to create the db are instance level variables. Each test method and the db creation method use those parameters to communicate indirectly about what db to create. This seems a strange pattern. What about removing those instance level variables, and pass them in directly in the test methods?

Hi @tuliren . Sure, I'm currently still in progress. During my work, I had to change the initial idea. Just want to finish with a few other connectors to see what else would I need to change and then update the structure accordingly. 
Thanks!

**Review by alexandr-shegeda — APPROVED** [review]
LGTM

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourceFillDbWithTestData.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        .format(getInsertQueryTemplate(), dbSchemaName, "%s", fieldsNames.toString(),
+            insertGroupValuesJoiner.toString());
+  }
+
+  // ex. INSERT INTO "test100tables100recordsDb".test_0 (id,test_column0)
+  // VALUES (0,'Value t0'), (1,'Value t1');
+  protected String prepareInsertQueryTemplatePostgres(final String dbSchemaName,
+                                                      final int batchNumber,
+                                                      final int numberOfColumns,
+                                                      final int recordsNumber) {
```
Comment (tuliren) [inline airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourceFillDbWithTestData.java:90]: This method is used not only by Postgres, but by MSSQL as well. So its name can be more generic.

File: airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/FillMySqlTestDbScriptTest.java [hunk 2]
```diff
@@ -90,30 +84,32 @@ protected String getTestFieldType() {
...
-   * @throws Exception might raise exception if configuration goes wrong or tables creation/insert
-   *         scripts failed.
+   * The test added test data to a new DB and check results. 1. Set DB creds in static variables above
+   * 2. Set desired number for streams, coolumns and records 3. Run the test
    */
-  private void setupDatabaseInternal() throws Exception {
-    final Database database = setupDatabase();
+  @Test
+  @Disabled
+  public void addTestDataToDbAndCheckTheResult() throws Exception {
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/FillMySqlTestDbScriptTest.java:93]: This method for MySQL is almost the same as those for Postgres and MSSQL. If a `numberOfBatches` is added to this method, I think all three databases can share the same method to insert the data, and this method can be lifted to the upstream class. In this way, the three classes should be more straightforward to read.

File: airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  public void test100tables100recordsDb() throws Exception {
+    int numberOfDummyRecords = 100;
+    int numberOfColumns = 240;
+    int numberOfStreams = 100;
+    String defaultDbSchemaName = "dbo";
+
+    setupDatabase("test100tables100recordsDb");
+
+    performTest(defaultDbSchemaName, numberOfStreams, numberOfColumns, numberOfDummyRecords);
+  }
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java:57]: All these test methods are the same / almost the same across the three databases. Can we move them to the upstream `AbstractSourcePerformanceTest` class as well? In that way, it is much simpler to add or remove a test.

File: airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/MySqlRdsSourcePerformanceSecretTest.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    String schemaName = "newinterim15tableswith50000records";
+
+    setupDatabase(schemaName);
+
+    performTest(schemaName, numberOfStreams, numberOfColumns, numberOfDummyRecords);
+  }
+
+  @Test
+  public void testRegular25tables50000recordsDb() throws Exception {
+    int numberOfDummyRecords = 50003;
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/MySqlRdsSourcePerformanceSecretTest.java:129]: Some of these numbers have slight differences. Is that on purpose? Not an issue. Just curious.

**Review by tuliren — COMMENTED** [review]
This is awesome!

----

I have two main categories of comments:

1. I think the classes that fill the data for the three databases can be simplified and DRYed a lot. Currently they have lots of duplicated code and are a bit confusing. The `addTestDataToDbAndCheckTheResult` can be unified and may be moved to the upstream `AbstractSourceFillDbWithTestData` class.

2. The performance test classes can probably be DRYed as well. If we move all the test methods to the upstream abstract class, each individual database test class no longer needs to duplicate the same test logic. And we can easily add or remove a test. The pattern is similar to the `DestinationAcceptanceTest`.

----

In addition, here are the current benchmark databases:

**MySQL and MSSQL**
| ID | Tables | Columns | Records | 
| --- | --- | --- | --- |
| D1 | 100 | 240 | 100 |
| D2 | 1K | 240 | 200 |
| D3 | 5K | 240 | 200 |
| D4 | 1K | 240 | 10K |
| D5 | 15 | 240 | 50K |
| D6 | 25 | 240 | 50K |

**Postgres**
| ID | Tables | Columns | Records | 
| --- | --- | --- | --- |
| D1 | 100 | 240 | 100 |
| D2 | 1K | 240 | 200 |
| D3 | 5K | 240 | 200 |
| D4 | 1K | 8 | 10K |
| D5 | 15 | 8 | 50K |
| D6 | 25 | 8 | 50K |

The difference in the number of columns (240 for MySQL and MSSQL, and 8 for Postgres) for D5 and D6 is probably a typo, right?

In addition to that, I think we can reduce the number of benchmarks:
1. D1 is very small, and it is covered by the other databases. I think it can be removed completely.
2. D5 and D6 are not that different. What about keeping D6, and delete D5?
3. D2 and D3 are trying to test the same thing: databases with a large number of tables. Can we just keep one of them?

In summary about the benchmark databases, I think only two are needed: one with small number of streams, with lots of records (e.g. D6), and one with lots of streams, but not many records (e.g. D2).

Maybe we can also keep D4 as a test for both cases. But we can increase the size of D6 and D2 to be closer to D4 in the future.

What do you think?

**Comment by tuliren** [comment]
Another thought indirectly related to this PR. Method `AbstractSourceConnectorTest#runReadVerifyNumberOfReceivedMsgs` mimics `DefaultReplicationWorker#getReplicationRunnable` to test the source connector.

1. It is vital that the former follows the same pattern as the latter. Is there a way to ensure that?
2. The latter method currently uses the `sourceMessageTracker` to track the stats of a sync on the source side. Ideally the former method should do the same.

File: airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/MySqlRdsSourcePerformanceSecretTest.java (hunk 4 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/MySqlRdsSourcePerformanceSecretTest.java:129]: Hi @tuliren. It depends on SQL script created by @andriikorotkov. As far as I understand - some PL\SQL differences between the DBs.

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/SourceBasePerformanceTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.standardtest.source.performancetest;
+
+import io.airbyte.integrations.standardtest.source.AbstractSourceConnectorTest;
+import io.airbyte.integrations.standardtest.source.TestDestinationEnv;
+
+public abstract class SourceBasePerformanceTest extends AbstractSourceConnectorTest {
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/SourceBasePerformanceTest.java:10]: Hi @tuliren
The AbstractSourceConnectorTest is the basic class for all kind of tests. The SourceBasePerformanceTest (I renamed it to AbstractSourceBasePerformanceTest) contains a few, but basic parts for both FillData scripts and Performance tests themselves. And the AbstractSourcePerformanceTest contains the basic part for Performance tests only.
I may get rid of AbstractSourceBasePerformanceTest, but then we will have some minor duplication in abstract classes for FillDb scripts and for Performance tests. 

What do you think?

File: airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java (hunk 3 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java]: Updated, thanks!

File: airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/FillMySqlTestDbScriptTest.java (hunk 2 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/FillMySqlTestDbScriptTest.java]: Updated, thanks!

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourceFillDbWithTestData.java (hunk 1 above)
Comment (etsybaev) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourceFillDbWithTestData.java]: Updated, thanks!

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourceFillDbWithTestData.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    return String.format(getCreateTableTemplate(), dbSchemaName, currentTableName, sj.toString());
+  }
+
+  // ex. INSERT INTO test.test_1 (id, test_column0, test_column1) VALUES (101,"zzz0", "sss0"), ("102",
+  // "zzzz1", "sss1");
+  protected String prepareInsertQueryTemplate(final String dbSchemaName,
+                                              final int numberOfColumns,
+                                              final int recordsNumber) {
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourceFillDbWithTestData.java:106]: Updated, thanks!

File: airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/FillMsSqlTestDbScriptTest.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FillMsSqlTestDbScriptTest extends AbstractSourceFillDbWithTestData {
+
+  protected static final Logger c = LoggerFactory.getLogger(FillMsSqlTestDbScriptTest.class);
+
+  private static final String CREATE_DB_TABLE_TEMPLATE = "CREATE TABLE %s.%s(id INTEGER PRIMARY KEY, %s)";
+  private static final String INSERT_INTO_DB_TABLE_QUERY_TEMPLATE = "INSERT INTO %s.%s (%s) VALUES %s";
+  private static final String TEST_DB_FIELD_TYPE = "varchar(10)";
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/FillMsSqlTestDbScriptTest.java]: Updated, thanks!

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourceFillDbWithTestData.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.standardtest.source.performancetest;
+
+import io.airbyte.db.Database;
+import java.util.StringJoiner;
+
+public abstract class AbstractSourceFillDbWithTestData extends SourceBasePerformanceTest {
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourceFillDbWithTestData.java]: Updated, thanks!

**Comment by etsybaev** [comment]
> Another thought indirectly related to this PR. Method `AbstractSourceConnectorTest#runReadVerifyNumberOfReceivedMsgs` mimics `DefaultReplicationWorker#getReplicationRunnable` to test the source connector.
> 
> 1. It is vital that the former follows the same pattern as the latter. Is there a way to ensure that?
> 2. The latter method currently uses the `sourceMessageTracker` to track the stats of a sync on the source side. Ideally the former method should do the same.

Hi @tuliren .
If I got you correctly - in the first case, we collect all received messages to collection and then verify that received messages are exactly as was expected. Meanwhile, such a direct approach wouldn't work for performance tests.  We will quickly end up with outOfMemory (for the test itself thread, not for connector's) error if try to collect all messages to a collection. In the case of performance tests, we are not interested in how properly it was read, but we care about the stream name and number of messages. So the former approach checks more carefully but doesn't work as it is for cases with big DBs. 
Hope that was what you asked for. Please let me know if I got your question incorrectly.
Thanks

**Comment by andriikorotkov** [comment]
@tuliren, I work on being able to pass CPU and memory values as gradle parameters for performance tests. for this I used this pull request from @etsybaev. I added my minor changes to this pull request, for _buildSrc / src / main / groovy / airbyte-performance-test-java.gradle_ and _airbyte-integrations / bases / standard-source-test / src / main / java / io / airbyte /integrations/standardtest/source/AbstractSourceConnectorTest.java_

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/SourceBasePerformanceTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (tuliren) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/SourceBasePerformanceTest.java]: I see. In that case, we can leave them as is. The new javadoc you added to the `AbstractSourceBasePerformanceTest` class is helpful.

File: airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+   * that will be ised as a NameSpace in Configured Airbyte Catalog. 3rd arg - a number of expected
+   * records retrieved in each stream. 4th arg - a number of columns in each stream\table that will be
+   * use for Airbyte Cataloq configuration 5th arg - a number of streams to read in configured airbyte
+   * Catalog. Each stream\table in DB should be names like "test_0", "test_1",..., test_n.
+   */
+  @BeforeAll
+  public static void beforeAll() {
+    AbstractSourcePerformanceTest.testArgs = Stream.of(
+        Arguments.of("test1000tables240columns200recordsDb", "dbo", 200, 240, 1000),
+        Arguments.of("test5000tables240columns200recordsDb", "dbo", 200, 240, 1000),
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java:61]: The `1000` here is a typo, right? Should be `5000` based on the database name.

Same thing for the other two databases.

File: airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        Arguments.of("newregular25tables50000records", "dbo", 50052, 8, 25),
+        Arguments.of("newsmall1000tableswith10000rows", "dbo", 10011, 8, 1000));
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java:63]: @andriikorotkov, there are some extra records in the benchmark tables due to database specific differences. Can you add some comments about why there are 10011 records instead exact 10K here? These numbers are totally fine. But they would be confusing without any explanation.

File: airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+  private JsonNode config;
+  private static final String PERFORMANCE_SECRET_CREDS = "secrets/performance-config.json";
+
+  @Override
+  protected JsonNode getConfig() {
+    return config;
+  }
+
+  @Override
+  protected void tearDown(final TestDestinationEnv testEnv) {}
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java:29]: I think the `config` variable, and the `getConfig` and `tearDown` methods can all be lifted to the `AbstractSourcePerformanceTest`.

**Review by tuliren — APPROVED** [review]
I see that there are still four databases we are testing against:

| ID | Tables | Columns | Records | Purpose |
| --- | --- | --- | --- | --- |
| D2 | 1K | 240 | 200 | Many tables. |
| D3 | 5K | 240 | 200 | Lots of tables. |
| D4 | 1K | 8 | 10K | Many tables + moderate records. |
| D6 | 25 | 8 | 50K | Lots of records. |

The cloud cannot handle 5000 tables currently. It fails at retrieving the catalog. Let's drop D3, and only run against D2 for now.

**Comment by tuliren** [comment]
> > Another thought indirectly related to this PR. Method `AbstractSourceConnectorTest#runReadVerifyNumberOfReceivedMsgs` mimics `DefaultReplicationWorker#getReplicationRunnable` to test the source connector.
> > 
> > 1. It is vital that the former follows the same pattern as the latter. Is there a way to ensure that?
> > 2. The latter method currently uses the `sourceMessageTracker` to track the stats of a sync on the source side. Ideally the former method should do the same.
> 
> Hi @tuliren . If I got you correctly - in the first case, we collect all received messages to collection and then verify that received messages are exactly as was expected. Meanwhile, such a direct approach wouldn't work for performance tests. We will quickly end up with outOfMemory (for the test itself thread, not for connector's) error if try to collect all messages to a collection. In the case of performance tests, we are not interested in how properly it was read, but we care about the stream name and number of messages. So the former approach checks more carefully but doesn't work as it is for cases with big DBs. Hope that was what you asked for. Please let me know if I got your question incorrectly. Thanks

I guess we are probably talking about different things. My first bullet point means the performance test should mimics the how the source is read as much as possible. Currently, we do that by replicating the production procedure (`DefaultReplicationWorker#getReplicationRunnable`) in `AbstractSourceConnectorTest#runReadVerifyNumberOfReceivedMsgs`. However, if the production procedure changes, the performance will not use the new procedure. So the test cannot truly reflect what happens in production. Ideally, the performance test should reference some of the production procedure directly, so that it is change-proof.

File: airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java (hunk 10 above)
Comment (andriikorotkov) [reply airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java]: @tuliren, this did not work correctly, due to the fact that different syntaxes do not allow inserting a large amount of records at one time. For example MsSQL, does not allow making an insert command with more than 1000 entries. I also found a small bug in the script and fixed it. Now each script inserts the number of records specified by the user. If you need 10000, then the script will add 10000 and not 10011 as it was before. i changed these numbers in the integration tests.
this did not work correctly, due to the fact that different syntaxes do not allow inserting a large amount of records at one time. For example MsSQL, does not allow making an insert command with more than 1000 entries. I also found a small bug in the script and fixed it. Now each script inserts the number of records specified by the user. If you need 10000, then the script will add 10000 and not 10011 as it was before. i changed these numbers in the integration tests.

Comment (tuliren) [reply airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java]: Cool. Thank you.

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourcePerformanceTest.java [hunk 12]
```diff
@@ -21,99 +20,79 @@
...
-  /**
-   * The stream name template will be used for a test tables. Override it if default name is not valid
-   * for your source.
-   *
-   * @return Test steam name template
-   */
-  protected String getTestStreamNameTemplate() {
-    return "test_%S";
-  }
+  protected static Stream testArgs;
```
Comment (tuliren) [inline airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourcePerformanceTest.java:37]:
```suggestion
  protected static Stream<Arguments> testArgs;
```

File: airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java (hunk 11 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java]: Updated, thanks!

File: airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java (hunk 9 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java]: You're right. We just had a broken Ci, so haven't tested it yet after refactor. Updated, thanks!

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourcePerformanceTest.java (hunk 12 above)
Comment (etsybaev) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourcePerformanceTest.java]: Updated, thanks!
